packages/webview_flutter/webview_flutter_android/lib/src/android_webview.dart

```diff
@@ -397,6 +404,11 @@ class WebView extends JavaObject {
     return api.setBackgroundColorFromInstance(this, color.value);
   }
 
+  ///Toggle scroll listener for this WebView
+  Future<void> enableContentOffsetChangedListener(bool enabled) {
```

Comment:
As mentioned in https://github.com/flutter/plugins/pull/2107#discussion_r1101579638, `WebView` should follow the pattern of `WebViewClient` for callback methods. A `WebViewFlutterApiImpl.java` has been added, so you should be able to add callbacks methods more easily now. You have already done the step of adding the `onScrollChanged` field to this class and implemented the method in the Dart `WebViewFlutterApiImpl`. You can remove this method, `enableContentOffsetChangedListener`, and instead 

1. Add the `onScrollChanged` method to the `WebViewFlutterApi` in `pigeons/android_webview.dart`
2.  Add an implementation of this callback in `WebViewFlutterApiImpl.java`. Something like
```java
public void onScrollChanged(@NonNull WebView instance, long left, long top, long oldLeft, long oldTop, @NonNull WebViewFlutterApi.Reply<Void> callback) {
      api.create(Objects.requireNonnull(instanceManager.getIdentifierForStrongReference(instance)), left, top, oldLeft, oldTop, callback);
  }
```
3. In `WebViewHostApiImpl.java`, you can instantiate a `WebViewFlutterApiImpl` in the overridden method of `WebViewPlatformView.onScrollChaged` and call the method created in step 2.

I'm still working on proper documentation for this design, so feel free to ask for more clarification on this.